sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java

```diff
@@ -426,26 +431,33 @@ private static FieldType fieldTypeFromProtoWithoutNullable(SchemaApi.FieldType p
           return FieldType.DATETIME;
         } else if (urn.equals(URN_BEAM_LOGICAL_DECIMAL)) {
           return FieldType.DECIMAL;
-        } else if (urn.equals(URN_BEAM_LOGICAL_JAVASDK)) {
-          return FieldType.logicalType(
-              (LogicalType)
-                  SerializableUtils.deserializeFromByteArray(
-                      logicalType.getPayload().toByteArray(), "logicalType"));
-        } else {
-          @Nullable FieldType argumentType = null;
-          @Nullable Object argumentValue = null;
-          if (logicalType.hasArgumentType()) {
-            argumentType = fieldTypeFromProto(logicalType.getArgumentType());
-            argumentValue = fieldValueFromProto(argumentType, logicalType.getArgument());
+        } else if (urn.startsWith("beam:logical_type:")) {
+          try {
+            return FieldType.logicalType(
+                (LogicalType)
+                    SerializableUtils.deserializeFromByteArray(
+                        logicalType.getPayload().toByteArray(), "logicalType"));
+          } catch (IllegalArgumentException e) {
+            LOG.warn(
+                String.format(
```

Comment:
`LOG.warn("Unable to deserialize the logical type {} from proto. Mark as UnknownLogicalType", urn)`